Documentation/ProvenHeaders.md

```diff
@@ -0,0 +1,229 @@
+
+It's important to mention that other coinstake transaction data may be fake (non-kernel inputs, fees claimed) but we don't care about this since this data is not related to the chain work. 
+
+
+
+#### Trusted base
+
+We define `trustedBase` as the block up until the node has UTXO set for that it trusts (including that block). In the C# Stratis full node  `trustedBase`  is represented by the consensus tip, and in Stratis C# light wallet node it is currently not represented. More on how trusted base should be represented for the light wallet is in the [trusted base for the light wallet](#trusted-base-for-the-light-wallet) section.
+
+
+
+#### Long reorg protection
+
+PoS blockchains similar to Stratis need to have a protection against long reorganizations. Stratis protocol prevents the node from switching to another chain that would replace more than 500 blocks. We denote this parameter as `maxReorg`.
+
+
+
+## Structure of proven header
+
+Proven header consists of:
+
+1. Block header (in the C# node is represented by `BlockHeader` class) which contains merkle tree root.
+
+2. Block header signature (in the C# node it's represented by `BlockSignature` class), which is signed with the private key which corresponds to coinstake's Second output's public key.
+
+3. Its coinstake transaction (in the C# node it is represented by `Transaction` class). 
+
+4. Merkle proof that proves the coinstake tx is included in a block that is being represented by the provided header. 
+
+   In the C# node the merkle proof can be represented by the `PartialMerkleTree` class.
+
+
+
+## Assumptions
+
+This document is written under assumption that a softfork that changes `coinstakeAge` consensus parameter from 50 to 500 is activated for Stratis.
+
+Solution proposed in this document should be implemented for the C# nodes only when miners that control at least 51% of all stake adopted the softfork. 
+
+
+
+#### Reasons behind changing coinstakeAge
+
+If the `coinstakeAge < maxReorg` then it is possible to create a valid coinstake kernel which will use UTXO originated in a block within the bounds of long reorg protection. Such coinstake can't always be validated with a small proof.
+
+Alternative solutions with longer proofs have been explored as well but they always introduce new attack vectors and additional complexity to the code which makes those solutions inferior to just implementing syncing with the *inv* messages.
+
+ 
+
+
+## Changes required for the C# node
+
+In this section changes that will be required for the C# node to support syncing with *proven headers* are covered. 
+
+#### Handshake
+
+Also after the handshake is completed *sendprovhdr* message should be sent instead of *sendheaders*.
+
+*sendprovhdr* message is a message just as *sendheaders* is and it is just another message type to distinguish between those two. Also it will contain height from which we prefer having proven headers over normal headers.
```

Comment:
```suggestion
*sendprovhdr* message is a message just as *sendheaders* is and it is just another message type to distinguish between those two. It will also contain the height from which we prefer having proven headers over normal headers.
```